tests/utils.go

```diff
@@ -766,10 +765,6 @@ func BeforeTestSuitSetup(_ []byte) {
 	createNamespaces()
 	createServiceAccounts()
 
-	if IsRunningOnKindInfraIPv6() {
-		createPodPreset("fix-node-uuid", "fake-product-uuid", "virt-launcher", "/kind/product_uuid", "/sys/class/dmi/id/product_uuid")
-	}
-
```

Comment:
SRIOV lane uses kind based providers and since the nodes are containers they dont have uuid
this line handles this, the file `/kind/product_id` is created during cluster-up on the cluster nodes
I know that ipv6 is not supported anymore, but it is necessary to do some fine tunes on the sriov provider
and handling the uuid is part of it.
Could you please keep this?
We can simply use `IsRunningOnKindInfra` function

Reply:
But this never ran on SRIOV lanes at all. `IsRunningOnKindInfraIPv6` is checking a specific lane, which is the ipv6 one. And I think these patches only handled migration related actions, something we do not support at the moment.

Reply:
Currently this line createa the podPreset for ipv6 kind cluster only. sriov kind cluster won't have it.
If you want to add it to sriov cluster please do it on a separate PR.
I don't see a reason to do it on this PR.

Reply:
> But this never ran on SRIOV lanes at all. `IsRunningOnKindInfraIPv6` is checking a specific lane, which is the ipv6 one. And I think these patches only handled migration related actions, something we do not support at the moment.

I have the same recollection; this was introduced to enable the migration functional tests to succeed (without this, all nodes on kind based infra will have the same UUID, which cause libvirt to prevent migrations).

I am not aware of any migration tests over sriov interfaces.

Reply:
I want to configure UUID for sriov nodes to prevent network issues, from what I know it could affect how a mac address is generated and could lead to IP allocation conflicts (where there is DHCP server)
It is a problem that we did not have UUID in kind cluster nodes in the first place.

We create `/kind/product_uui` at each node from the provider script file ([for example](https://github.com/kubevirt/kubevirtci/blob/master/cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh#L30),
then create the PodPreset here (line 770) from kubevirt/kubvirt tests/utils.go.

So I would like to ask you to not remove the function `createPodPreset` and instead of using `IsRunningOnKindInfraIPv6` in line 769 use `IsRunningOnKindInfra` please, so I wont need to add this code again :)

Reply:
The removal of this code has been reasoned by removing the IPv6 kind lane. The reasoning is strong and hard to argue against.

Your reasoning here is also good, but it has no proof as there is no apparent problem at the moment.
The history is not gone, so you can reintroduce this when you have a real need to bring it in. If your arguments are correct, it should be no problem to recreate it easily and present it through a failing test to convince the reviewers.

Reply:
Why wait for a problem to appear? 
Its not about removing IPv6 lane, its just removing `IsRunningOnKindInfraIPv6` function. 
The function that should have been used in the first place is `IsRunningOnKindInfra`, 
so it would affect any provider that use kind and not only the IPv6 one, it was missed.
There is no doubt that its something that should be configured on the nodes.

Reply:
> Why wait for a problem to appear? There is no doubt that its something that should be configured on the nodes.

Well ... It is **if** you want migration tests over sriov - which don't exist - to succeed.

I guess this means it can go either way. 

> 
> Its not about removing IPv6 lane, its just removing `IsRunningOnKindInfraIPv6` function.
> The function that should have been used in the first place is `IsRunningOnKindInfra`,

+1 ; that was discussed [here](https://github.com/kubevirt/kubevirt/pull/3374#discussion_r420060046). 

> so it would affect any provider that use kind and not only the IPv6 one, it was missed.